lxd/network/driver_common.go

```diff
@@ -209,6 +209,113 @@ func (n *common) validateZoneNames(config map[string]string) error {
 	return nil
 }
 
+// validateRoutes checks that ip routes are compatible with existing forwards and load balancers.
+func (n *common) validateRoutes(config map[string]string) error {
+	var (
+		routesListIPv4 []*net.IPNet
+		routesListIPv6 []*net.IPNet
+
+		forwards      map[string]map[string][]string
+		loadBalancers map[string]map[string][]string
+
+		err error
+	)
+
+	if config["ipv4.routes"] != "" {
+		routesListIPv4, err = shared.ParseNetworks(config["ipv4.routes"])
+		if err != nil {
+			return fmt.Errorf("Failed parsing ipv4.routes: %w", err)
+		}
+	}
+
+	if config["ipv6.routes"] != "" {
+		routesListIPv6, err = shared.ParseNetworks(config["ipv6.routes"])
+		if err != nil {
+			return fmt.Errorf("Failed parsing ipv6.routes: %w", err)
+		}
+	}
+
+	// Get all listen addresses for all dependent OVN networks.
+	// OVN networks do not support per-member forwards.
+	err = n.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+		forwards, err = tx.GetProjectNetworkForwardListenAddressesByUplink(ctx, n.name, false)
+		if err != nil {
+			return fmt.Errorf("Failed to get listen addresses of OVN network forwards: %w", err)
+		}
+
+		loadBalancers, err = tx.GetProjectNetworkLoadBalancerListenAddressesByUplink(ctx, n.name, false)
+		if err != nil {
+			return fmt.Errorf("Failed to get listen addresses of OVN load balancers: %w", err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	checkListenAddressesOVN := func(project, network string, ips []string) error {
+		for _, ip := range ips {
+			var routesKey string
+
+			netIP := net.ParseIP(ip)
+			routeExists := false
+
+			if netIP.To4() != nil {
+				// Listen address is IPv4.
+				for _, routes := range routesListIPv4 {
+					if routes.Contains(netIP) {
+						routeExists = true
+					}
+				}
+
+				routesKey = "ipv4.routes"
+			} else {
+				// Listen address is IPv6.
+				for _, routes := range routesListIPv6 {
+					if routes.Contains(netIP) {
+						routeExists = true
+					}
+				}
+
+				routesKey = "ipv6.routes"
+			}
+
+			if !routeExists {
+				return fmt.Errorf("%q is missing network for listener %q of network %q in project %q", routesKey, ip, network, project)
+			}
+		}
+
+		return nil
+	}
+
+	// Check that IP routes are provided for already existing OVN forwards.
+	for project, networks := range forwards {
+		for network, ips := range networks {
+			if n.name == network && n.project == project {
```

Comment:
as this check would be safe to do for load balancers as well, it seems like we could combine these two sections for forwards and loadBalancers into a single look, e.g.

```go
for _, routes := range []{forwards, loadBalancers} {
    for project, networks := range routes {
        ....
    }
}
```

```